tools/mtmd/clip.cpp

```diff
@@ -1715,6 +1509,34 @@ struct clip_graph {
         return gf;
     }
 
+    ggml_cgraph * build_janus_pro() {
+        GGML_ASSERT(model.class_embedding == nullptr); // No CLS token
+
+        ggml_tensor * inp = build_inp();
+
+        ggml_tensor * learned_pos_embd = model.position_embeddings;
+
+        ggml_tensor * cur = build_vit(
+                                inp, n_patches,
+                                NORM_TYPE_NORMAL,
+                                hparams.ffn_op,
+                                learned_pos_embd,
+                                nullptr);
+
+        cur = build_ffn(cur,
+            model.mm_0_w, model.mm_0_b,
+            nullptr, nullptr,
+            model.mm_1_w, model.mm_1_b,
+            hparams.ffn_op,
+            -1);
+        cb(cur, "aligner_1", -1);
+
+        // build the graph
+        ggml_build_forward_expand(gf, cur);
+
+        return gf;
+    }
```

Comment:
I think we can now merge this function with `build_siglip()`, you only need to add another `else if (ctx->proj_type() == ...)`